examples/generator/mock_to_qir.py

```diff
@@ -40,15 +41,27 @@ def write_to_file(self, file_path: str) -> str:
             file.write(self.ir_string)
 
     def enterXGate(self, ctx: MockLanguageParser.XGateContext):
+        if int(ctx.target.text) >= self.nr_qubits:
+            raise ValueError("Parsed progam uses more qubits than allocated")
```

Comment:
Maybe you could factor out this check into a function?

Reply:
@samarsha Thanks for the push. Updated it.